source/common/protobuf/utility.cc

```diff
@@ -261,6 +275,22 @@ void ProtoExceptionUtil::throwProtoValidationException(const std::string& valida
   throw ProtoValidationException(validation_error, message);
 }
 
+// TODO(htuch): this is where we will also reject v2 configs by default.
+void MessageUtil::onVersionUpgrade(absl::string_view desc) {
+  const std::string& warning_str =
+      fmt::format("Configuration does not parse cleanly as v3. v2 configuration is "
+                  "deprecated and will be removed from Envoy at the start of Q1 2021: {}",
+                  desc);
+  ENVOY_LOG_MISC(trace, warning_str);
+  ENVOY_LOG_PERIODIC_MISC(warn, 1s, warning_str);
```

Comment:
Is this going to be useful for the user since they may not see all the things that are warning? It's kind of awful but should this be a static map of warnings and we log each unique one occasionally? Thoughts?

Reply:
I added a static map, LMK what you think.